UnitTest/mocks.c

```diff
@@ -0,0 +1,27 @@
+#include <stdio.h>
+#include <stdarg.h>
+#include "mocks.h"
+
+int PrintfInputParam[3];
+int call_printf= 0;
+
+int stub_printf(const char *format , ...)
```

Comment:
sorry for naming it wrongly. Its a mock as you taught us.
(copy pasting issues.)
This indicates an opaque design of the code, without visibility into its working. --> did not understood it could you please add some more details @sudeeprp